src/java.base/share/classes/java/lang/reflect/code/op/ExtendedOp.java

```diff
@@ -3158,6 +3158,9 @@ static Block.Builder lowerTypePattern(Block.Builder endNoMatchBlock, Block.Build
                 boolean patternWithPrimitive = false;
                 if (targetType instanceof PrimitiveType tt && target.type() instanceof PrimitiveType st) {
                     patternWithPrimitive = true;
+                    if (List.of(CHAR, SHORT).contains(st)) {
```

Comment:
This idiom is ok for tests, but generally it not very efficient. Just do `if (CHAR.equals(st) || SHORT.equals(st)) { ..` and prefix with a comment.